File: 101-109_small_problems/easy_9/welcome_stranger.rb
# Welcome Stranger
# Create a method that takes 2 arguments, an array and a hash. The array will contain 2 or more elements that, when combined with adjoining spaces, will produce a person's name. The hash will contain two keys, :title and :occupation, and the appropriate values. Your method should return a greeting that uses the person's full name, and mentions the person's title.

# Input: An array and a hash
# Output: A string
# Mental model: Given an array and a hash, return a string containing elements from input.

# Example:

# greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' })
# => Hello, John Q Doe! Nice to have a Master Plumber around.

# Data structure

# Input is an array and a hash and output is a string.

# Algorithm

# Create a new string with string interpolation with each element of the array and each element of the hash.

# Code

def greetings(arr, hsh)
  print "Hello #{arr.join(" ")}! "
  print "Nice to have a #{hsh[:title]} #{hsh[:occupation]} around."
end

greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' })

# Further Exploration
# This method doesn't actually pass rubocop! What is the best way to shorten the lines in this method so it doesn't exceed the 80 maximum characters to a line?

File: lessons/210_lessons/lesson_5/welcomeStranger.js
// Practice Problem: Welcome Stranger
// Write a function that takes two arguments, an array and an object. The array will contain two or more elements that, when combined with spaces, produce a person's full name. The object will contain two keys, title and occupation, and suitable values for both items. Your function should log a greeting to the console that uses the person's full name, and mentions the person's title and occupation.
// Input: An array and an object
// Output: A string

// Algorithm
// Join array into a string
// Create a new string for occupation
// Add values to occupation
// Return string: 'Hello, ' + name + '! Nice to have a ' + occupation + ' around.'

function greetings(name, titleAndOccupation) {
  var fullName = name.join(' ');
  var job = titleAndOccupation.title + ' ' + titleAndOccupation.occupation;

  console.log('Hello, ' + fullName + '! Nice to have a ' + job + ' around.');
}

// Example:

greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' });

// console output
// Hello, John Q Doe! Nice to have a Master Plumber around.
